providers/base/bin/network.py

```diff
@@ -210,9 +210,13 @@ def extract_core_list(self, line):
         """Extract a list of CPU cores from a line of the form:
         NUMA node# CPU(s):    a-b[,c-d[,...]]"""
         colon = line.find(":")
-        cpu_list = line[colon + 1 :]
+        cpu_list = line[colon + 1:]
         core_list = []
         for core_range in cpu_list.split(","):
+            # Skip it if the CPU list for the NUMA node is empty....
+            core_range = core_range.strip()
+            if not core_range:
+                continue
```

Comment:
I'd recommend you add a log message here:


```suggestion
            if not core_range:
                logging.error("Empty core range in line: %s", line)
                continue
```